include/zmq.h

```diff
@@ -463,14 +463,16 @@ ZMQ_EXPORT int zmq_socket_monitor (void *s, const char *addr, int events);
 #define ZMQ_POLLERR 4
 #define ZMQ_POLLPRI 8
 
-typedef struct zmq_pollitem_t
-{
-    void *socket;
 #if defined _WIN32
-    SOCKET fd;
+typedef SOCKET zmq_fd_t;
```

Comment:
I'm not sure I like the idea of adding a new typedef to the ABI, especially changing existing ones... it's fine for internal usage ofc

Reply:
Why should a typedef change the ABI?

Reply:
I meant changing existing definitions - with all the broken compilers out there (*cough*oracle*cough*) I'd really rather avoid the risk of changing existing structs when there's no overwhelming benefits

Reply:
Hm. Ok. I will revert the changes to zmq.h